### PR TITLE
Implement `Display`; add simple binary

### PR DESCRIPTION
**Pull request by fitzgen**
See each commit for details.

### CODE REVIEW

**Comment by brayniac** [comment]
Awesome! This looks pretty useful to have. Give me some time to more thoroughly review and play with it. Do you have any samples?

**Comment by fitzgen** [comment]
I'm on my phone now, but the biggest thing hurting usability when I was
playing with it was the lack of an ability to say "only use n buckets"

**Comment by brayniac** [comment]
It might be required to specialize a non-"hdrhistogram-style" structure for that? I would need to think about it more - but the bucket requirements are based on range of values and how many sig-figs are required. Not sure if there's an intuitive way to combine that promise of precision with a "use this many buckets exactly" requirement.

File: src/lib.rs [hunk 1]
```diff
@@ -319,6 +320,55 @@ impl fmt::Debug for Histogram {
...
+        writeln!(f, "# Each ∎ is a count of {}", count_per_char)?;
+        writeln!(f, "#")?;
+
+        let mut buckets: Vec<_> = self.into_iter().skip_while(|b| b.count == 0).collect();
+        while buckets.last().map_or(false, |b| b.count == 0) {
+            buckets.pop();
+        }
+
+        for Bucket { count, value, width, .. } in buckets {
+            write!(f, "{:>10} - {:>10} [ {:>10} ]: ", value, value + width, count)?;
```
Comment (brayniac) [inline src/lib.rs]: This needs to be:
```
write!(f, "{:>10} - {:>10} [ {:>10} ]: ", value - width + 1, value, count)?;
```
The value is the upper bound of the bucket - not the lower bound. Intent there is to slightly overstate latency percentiles. Changing the formula here will work for both the linear and non-linear portions of the histogram

File: src/bin/histogram.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+fn main() {
+    if let Err(e) = try_main() {
+        let mut stderr = io::stderr();
+        let _ = write!(&mut stderr, "error: {}", e);
+        process::exit(1);
+    }
+}
+
+fn try_main() -> io::Result<()> {
+    let mut hist = histogram::Histogram::default();
```
Comment (brayniac) [inline src/bin/histogram.rs]: Probably out of scope for this PR, but I would consider giving the user flexibility of setting a precision other than the default - for a range of 1-1M I wind up with 188 lines out with precision = 1 but with precision = 3 (default) it's 11k lines. Both are probably too much for terminal output. Being able to set a max value for the expected input may also help constrain the number of buckets. 

Feel free to ignore for this PR.

**Comment by fitzgen** [comment]
Pushed a fix for the bucket ranges.

**Comment by brayniac** [comment]
The changes looked good. Still failing the travis-ci rustfmt pass on new code blocks. Please let me know if you intend to address that.

**Comment by fitzgen** [comment]
Ah, I missed the rustfmting. I'll let you take this PR to the finish line.

FWIW, I ended up writing my own dumb O(n) space histogram so I could control number of buckets: https://github.com/fitzgen/histo

Cheers!

**Comment by brayniac** [comment]
@fitzgen - sounds good. Thanks again for your submission.

**Comment by brayniac** [comment]
Closing this PR without merge. This would be pretty cool, but makes more sense in the context of its own lib.

**Pull request closed without merging**